### PR TITLE
General perfomance optimizations.

### PR DESCRIPTION
**Pull request by hb3p8**
After some profiling with callgrind I was able to find a few bottlenecks which may be easily fixed.

### CODE REVIEW

**Comment by wheybags** [comment]
development has moved to https://github.com/OpenMW/openmw
please submit this there

**Comment by hb3p8** [comment]
Do I need to refork repo in order to make pull request to https://github.com/OpenMW/openmw?

**Comment by ghost** [comment]
No, when submitting the pull request you just need to click on "Edit" at the top right and select OpenMW/openmw on the "Base fork" drop down.

**Comment by hb3p8** [comment]
Done

File: apps/openmw/mwrender/occlusionquery.cpp [hunk 1]
```diff
@@ -27,6 +27,9 @@
     mRendering = renderer;
     mSunNode = sunNode;
 
+    mSupported = false;
+    return;
```
Comment (jeaye) [inline apps/openmw/mwrender/occlusionquery.cpp:31]: Unconditional return here? That's a behavior change, not a perf optimization.

File: components/nifogre/controller.hpp [hunk 2]
```diff
@@ -36,16 +41,21 @@
             if(time <= keys.front().mTime)
                 return keys.front().mValue;
 
-            Nif::Vector3KeyList::VecType::const_iterator iter(keys.begin()+1);
-            for(;iter != keys.end();iter++)
+            const Nif::Vector3Key* keyArray = keys.data();
+            size_t size = keys.size();
+
+            for (size_t i = 1; i < size; ++i)
```
Comment (jeaye) [inline components/nifogre/controller.hpp:47]: The only benefit I see here is the caching of the escape condition, which could've simply been done with `Nif::Vector3KeyList::VecType::const_iterator const end(keys.end());` before the loop. Do you've any profiling results indicating iterators are slower than indices here? This seems more like a style change.

File: apps/openmw/mwrender/occlusionquery.cpp (hunk 1 above)
Comment (hb3p8) [inline apps/openmw/mwrender/occlusionquery.cpp:31]: This shouldn't be actually in commit. I have proposed fixed one to https://github.com/OpenMW/openmw repo.

File: components/nifogre/controller.hpp (hunk 2 above)
Comment (hb3p8) [inline components/nifogre/controller.hpp:47]: I think you are right and we can go back to iterators now. But caching indeed is important here. These interpolators are used intensively.

**Comment by zinnschlag** [comment]
Can we close this pull request?

**Comment by hb3p8** [comment]
Of course.

**Pull request closed without merging**